frontend/tests/components/date-picker-field.test.tsx

```diff
@@ -0,0 +1,143 @@
+import { createRoutesStub, useLocation } from 'react-router';
+
+import { render } from '@testing-library/react';
+import { describe, expect, it, vi } from 'vitest';
+
+import { DatePickerField } from '~/components/date-picker-field';
+
+vi.mock(import('react-router'), async (importOriginal) => {
+  const actual = await importOriginal();
+  return {
+    ...actual,
+    useLocation: vi.fn(),
+  };
+});
+
+describe('DatePickerField', () => {
+  vi.mocked(useLocation, { partial: true }).mockReturnValue({ pathname: '/en/some-path' });
```

Comment:
This shouldn't be required. If you need `useLocation()` to return a path with a language, just set it in the `<RoutesStub>`

Reply:
I just ran your tests without the `vi.mock()` and `vi.mocked()` and they pass.